### DISCUSSION TITLE
Add fonctionnality "external plugin", like a market place ?

### DISCUSSION THREAD

**Discussion by abarrau** [magicbug/Cloudlog#2518] [General]
Hello friends,
 
About 2 years ago, I started creating an add-on for Cloudlog, called "external plugin".

The idea being that each developer can create these own complementary developments without impacting the main core of cloudlog; a bit like a “market place”.

I didn't have time to formalize everything in a doc.
To date, I have 2 plugins completed and functional (the first of which has been running for 1 year (sharedata)):
- sharedata: allows you to share the latest QSOs (and put them in iframe on QRZ.com, for example); also indicates whether one is active or not.
-- https://github.com/abarrau/Cloudlog-plugins/wiki/Cloudlog-:-plugin-%22sharedata%22
- contest: complements cloudlog's "contest" functionality; it allows you to validate the QSO belonging to the contest, be able to have statistics, make an export in EDI format
-- https://github.com/abarrau/Cloudlog-plugins/wiki/Cloudlog-:-plugin-%22contest%22

Other modules are in progress (satellite tracking, search in REF nomenclature, task planner).

You can see the information here (how to install) : https://github.com/abarrau/Cloudlog-plugins/wiki/Cloudlog-:-external-plugins-:-add-fonctionnality

@magicbug Peter, would you be willing to add this functionality to Cloudlog? (5 files and 3 tables).
Do not hesitate to get back to me if you have any questions.

73 to all
Aurélien / F4ANS

**Comment by magicbug** [comment]
Hi Aurélien,

I love the idea of plugins for Cloudlog, I think we're getting to the point where core functionality is there now so addons should be explored now, not sure the best method of implementation but I'll take a look and open to feedback/ideas from users.

Peter

**Comment by abarrau** [comment]
Thanks for your feedback @magicbug Peter.

The basics are present in my development, cf my gith page (everything is ready :))
If this is done, I will just have to document the method of using the functionalities offered by this "external-plugin" module to allow others to do it as well.

Do not hesitate if you have questions.

**Comment by magicbug** [comment]
My big worry is portability of plugins, ideally, it would be nice to keep them completely separate in something like /application/plugins but not sure how practical that would be, but I do worry if we allow plugins to mix with the core functions if we are asking for trouble in the long term

**Comment by abarrau** [comment]
Hello Peter, 
The proposed software architecture is completely independent, with its own directory where the plugins have their own architecture (controller, view, language, etc.)
The data is stored in 3 tables formatted to accept the configurations of each plugin.
plugins can use core functions.
Each user can decide to install the plugin or not and in fact if the "external plugin" functionality is not desired, it can be deactivated via the configuration file

Available to exchange

**Comment by magicbug** [comment]
Just been having a play with it and overall looks good, I'd suggest submitting it as a PR if you're interested in having it added to Cloudlog so we can easily check it against the codebase and offer any ideas on changes that might be needed

**Comment by GregTroar** [comment]
Aurelien,

As I see you seem to code well in PHP would you think you could spend a bit of time to implement the DDFM Award for French department ?
Thanks
